spec/models/user_spec.rb

```diff
@@ -42,110 +42,130 @@
     end
   end
 
-  describe 'is_admin?' do
-    it 'should be true if the group is an admin group' do
-      allow(subject).to receive(:groups).and_return(['workgroup:sdr:administrator-role'])
-      expect(subject.is_admin?).to be true
+  describe '#is_admin?' do
+    subject { user.is_admin? }
+    let(:user) { described_class.new }
+    before do
+      allow(user).to receive(:groups).and_return(groups)
     end
-    it 'should be false if the group is a deprecated admin group' do
-      allow(subject).to receive(:groups).and_return(['workgroup:dlss:dor-admin'])
-      expect(subject.is_admin?).to be false
+
+    context 'when the group is a deprecated admin group' do
+      let(:groups) { ['workgroup:dlss:dor-admin'] }
+      it { is_expected.to be false }
     end
-    it 'should be false for a blank group' do
-      allow(subject).to receive(:groups).and_return([])
-      expect(subject.is_admin?).to be false
+
+    context 'when there are no groups' do
+      let(:groups) { [] }
+      it { is_expected.to be false }
     end
-    it 'should be false with an inadequate group membership' do
-      allow(subject).to receive(:groups).and_return(['workgroup:dlss:not-admin'])
-      expect(subject.is_admin?).to be false
+
+    context 'with an inadequate group membership' do
+      let(:groups) { ['workgroup:dlss:not-admin'] }
+      it { is_expected.to be false }
     end
-    it 'should be true for ADMIN_GROUPS' do
-      allow(subject).to receive(:groups).and_return(User::ADMIN_GROUPS)
-      expect(subject.is_admin?).to be true
+
+    context 'with ADMIN_GROUPS' do
+      let(:groups) { User::ADMIN_GROUPS }
+      it { is_expected.to be true }
     end
-    it 'should be false for MANAGER_GROUPS' do
-      allow(subject).to receive(:groups).and_return(User::MANAGER_GROUPS)
-      expect(subject.is_admin?).to be false
+
+    context 'with MANAGER_GROUPS' do
+      let(:groups) { User::MANAGER_GROUPS }
+      it { is_expected.to be false }
     end
-    it 'should be false for VIEWER_GROUPS' do
-      allow(subject).to receive(:groups).and_return(User::VIEWER_GROUPS)
-      expect(subject.is_admin?).to be false
+
+    context 'with VIEWER_GROUPS' do
+      let(:groups) { User::VIEWER_GROUPS }
+      it { is_expected.to be false }
     end
   end
 
-  describe 'is_webauth_admin?' do
-    it 'should be true for ADMIN_GROUPS' do
-      allow(subject).to receive(:webauth_groups).and_return(User::ADMIN_GROUPS)
-      expect(subject.is_webauth_admin?).to be true
+  describe '#is_webauth_admin?' do
+    subject { user.is_admin? }
+    let(:user) { described_class.new }
+    before do
+      allow(user).to receive(:webauth_groups).and_return(groups)
     end
 
-    it 'should be true when impersonating other group membership' do
-      allow(subject).to receive(:webauth_groups).and_return(User::ADMIN_GROUPS)
-      allow(subject).to receive(:groups).and_return(['workgroup:dlss:not-admin'])
```

Comment:
this change seems to lose the point of this test:  that the user is only in a non-admin group according to `#groups` (presumably due to impersonation of another group list).  but they are actually an admin according to `#webauth_groups` (this is what allows them to keep the link to re-set the groups to impersonate, even when they are impersonating a non-super-user).

Reply:
I'm not sure I follow. Here I'm only setting `webauth_groups`, and we see that when the `webauth_groups` have the admin group, then `is_webauth_admin?` returns `true`. Maybe your concern is addressed by #1075